go-controller/pkg/node/linkmanager/link_network_manager.go

```diff
@@ -0,0 +1,389 @@
+		if !addressSaved.Equal(address) {
+			temp = append(temp, address)
+		}
+	}
+	c.store[linkName] = temp
+}
+
+// GetAssignedAddressLabel returns the label that must be assigned to each egress IP address bound to an interface
+func GetAssignedAddressLabel(linkName string) string {
+	//TODO: undeterstand why label char limt is 15 and how can a label work with a long link name otherwise it may block eip assignment
+	return fmt.Sprintf("%sovn", linkName)
+}
+
+// GetExternallyAvailableNetlinkLinkAddresses gets all addresses assigned on an interface with the following characteristics:
+// Must be up
+// Must not be loopback
+// Must not have a parent
+// Address must have scope universe
+func GetExternallyAvailableNetlinkLinkAddresses(link netlink.Link, v4, v6, skipBridges bool) ([]netlink.Addr, error) {
+	validAddresses := make([]netlink.Addr, 0)
+	flags := link.Attrs().Flags.String()
+	// exclude loopback interfaces
+	if strings.Contains(flags, "loopback") {
+		return validAddresses, nil
+	}
+	// exclude interfaces that aren't up
+	if !strings.Contains(flags, "up") {
+		return validAddresses, nil
+	}
+	// skip any devices which have an owner or parent
+	if skipBridges {
+		if link.Attrs().MasterIndex != 0 {
+			return validAddresses, nil
+		}
+	}
+	// ditto
+	if link.Attrs().ParentIndex != 0 {
+		return validAddresses, nil
+	}
+	linkAddresses, err := getAllLinkAddressesByIPFamily(link, v4, v6)
+	if err != nil {
+		return validAddresses, fmt.Errorf("failed to get all valid link addresses: %v", err)
+	}
+	for _, address := range linkAddresses {
+		// consider only GLOBAL scope addresses
+		if address.Scope != int(netlink.SCOPE_UNIVERSE) {
+			continue
+		}
+		validAddresses = append(validAddresses, address)
+	}
+	return validAddresses, nil
+}
+
+// GetExternallyAvailableNetipAddresses returns address Prefixes from interfaces with the following characteristics:
+// Must be up
+// Must not be loopback
+// Must not have a parent
+// Address must have scope universe
+func GetExternallyAvailableNetipAddresses(link netlink.Link, v4, v6, skipBridges bool) ([]netip.Prefix, error) {
+	validAddresses := make([]netip.Prefix, 0)
+	flags := link.Attrs().Flags.String()
+	// exclude loopback interfaces
+	if strings.Contains(flags, "loopback") {
+		return validAddresses, nil
+	}
+	// exclude interfaces that aren't up
+	if !strings.Contains(flags, "up") {
+		return validAddresses, nil
+	}
+	// skip any devices which have an owner or parent
+	if skipBridges {
+		if link.Attrs().MasterIndex != 0 {
+			return validAddresses, nil
+		}
+	}
+	if link.Attrs().ParentIndex != 0 {
+		return validAddresses, nil
+	}
+	linkAddresses, err := getAllLinkAddressesByIPFamily(link, v4, v6)
+	if err != nil {
+		return validAddresses, fmt.Errorf("failed to get all valid link addresses: %v", err)
+	}
+	for _, address := range linkAddresses {
+		// consider only GLOBAL scope addresses
+		if address.Scope != int(netlink.SCOPE_UNIVERSE) {
+			continue
+		}
+		// skip Egress IPs
```

Comment:
this function is very similar to the previous function. I would consider consolidating them. Why do you skip egress ips here, but not in the previous function?

Reply:
Extracted some common logic and renamed the func / updated the func doc string to make it clearer the differences between the two funcs.

I skip egress IPs in one function because its used by consumers to find a link to assign an EIP to and I didnt want to loop over the addresses again and further filter. I can do this if you want though to increase readability.

We also need a function for the EIP reconcile engine to include the EIPs so we can detect if an EIP is already assigned.

Its hard to consolidate it further because consumer needs `netip.Prefix` type to build LPM tree. :/

Resolving this - feel free to dispute and suggest rework.